OpenSim/Common/Component.cpp

```diff
@@ -1103,6 +1107,29 @@ void Component::setNextSubcomponentInSystem(const Component& sub) const
 void Component::updateFromXMLNode(SimTK::Xml::Element& node, int versionNumber)
 {
     if (versionNumber < XMLDocument::getLatestVersion()) {
+        if (versionNumber < 30500) {
+            // In 3.3 and earlier, spaces in names were tolerated. Spaces are
+            // no longer acceptable in Component names.
```

Comment:
Perhaps include in CHANGELOG (this change could affect a user's downstream code and goes beyond "no need to update because this only adds error checking" IMHO).

Reply:
Good point, I'll add the stricter naming rules to the CHANGELOG. Old models will still load but components will be renamed for you!

Reply:
> Old models will still load but components will be renamed for you!

I know, but I thought the point of CHANGELOG (in theory, at least) was for it to describe all the changes a user would need to make to his/her pipeline if upgrading from 3.3 to 4.0. A user will need to update code if it refers to Components by name and if any of those names are now disallowed.

Reply:
The following was added to the CHANGELOG:

> Component naming is more strictly enforced and names with spaces are no longer accepted. Spaces are only allowable as separators for Output or Channel names that satisfy a list Input. (PR #1955)

I think it implies that you have to update your Component names wherever you use them. Would you prefer I state that explicitly?

Reply:
Spaces are allowed in any list property of type string, as delimiters between the list's entries. The value of list property XML elements is somewhat separate from what is allowed in the `name="..."` annotation.

Reply:
> I think it implies that you have to update your Component names wherever you use them. Would you prefer I state that explicitly?

In general, I would briefly describe what was changed and then tell the user what he/she needs to do. Perhaps add something like "OpenSim will automatically rename any Components whose names contain the following characters: …"?